### PR TITLE
[Defacepocalypse] Import products#edit and form partial in our codebase and apply overrides

### LINKED ISSUE #3051: [Defacepocalypse] Apply html.haml.deface overrides onto admin/products/new.html.haml
## Description

We need to apply in `master` the following list of overrides

```shell
$ ll app/overrides/spree/admin/products/_form
total 44K
-rw-r--r-- 1 pau pau  358 Nov 14 17:04 add_description_wysiwyg.html.haml.deface
-rw-r--r-- 1 pau pau  106 Nov 14 17:04 add_primary_taxon_field.html.haml.deface
-rw-r--r-- 1 pau pau  300 Nov 14 17:04 add_supplier.html.haml.deface
-rw-r--r-- 1 pau pau 1.1K Nov 14 17:04 add_units_form.html.haml.deface
-rw-r--r-- 1 pau pau  131 Nov 14 17:04 remove_available_on.deface
-rw-r--r-- 1 pau pau  121 Nov 14 17:04 remove_cost_currency.deface
-rw-r--r-- 1 pau pau  118 Nov 14 17:04 remove_cost_price.deface
-rw-r--r-- 1 pau pau   49 Nov 14 17:04 remove_meta_description.deface
-rw-r--r-- 1 pau pau   48 Nov 14 17:04 remove_option_types_and_taxons.deface
-rw-r--r-- 1 pau pau  172 Nov 14 17:04 replace_master_price_label.html.haml.deface
-rw-r--r-- 1 pau pau  197 Nov 14 17:04 replace_taxons_div.html.haml.deface
```
onto the `app/views/spree/admin/products/new.html.haml` view that results from https://github.com/openfoodfoundation/openfoodnetwork/issues/3050. Most of them are a single-line overrides.

We also have to be sure that the following overrides

```shell
$ ll app/overrides/spree/admin/products/edit 
total 12K
-rw-r--r-- 1 pau pau  82 Nov 14 17:04 add_angular.deface
-rw-r--r-- 1 pau pau 142 Jun 11 14:32 nojs_new_product.html.haml.deface
-rw-r--r-- 1 pau pau 189 Nov 14 17:04 return_to_products.html.haml.deface
```

work after the change, as well as the edit page itself.

### PR DESCRIPTION
**Pull request by HugsDaniel**
#### What? Why?

Closes #3051 

Import products#edit and form partial in our codebase and apply overrides.

#### What should we test?

The products edit page should be working fine.
Also, the field "Product Category" should now appear. The deface file making this happen pointed to div which is not in Spree, but is added with deface later on. As the div wasn't here yet it didn't appear on the page.

#### Release notes

Import products#edit and form partial in our codebase 

Changelog Category: Added
#### How is this related to the Spree upgrade?

This is another Defacepocalypse step that will help us when going to Spree 2.0

### CODE REVIEW

File: app/views/spree/admin/products/_form.html.haml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      = f.collection_select(:tax_category_id, @tax_categories, :id, :name, { :include_blank => 'None' }, { :class => 'select2' })
+      = f.error_message_on :tax_category
+
+  .clear
+
+  %div
+
+  .clear
+
+- unless Rails.env.test?
```
Comment (sauloperez) [inline app/views/spree/admin/products/_form.html.haml]: This :nose: bad... but this is Spree's I guess, right?

Comment (HugsDaniel) [reply app/views/spree/admin/products/_form.html.haml]: Right :/

**Comment by sauloperez** [comment]
Lol @HugsDaniel I think that last ammend you did didn't work as expected :trollface:

File: app/views/spree/admin/products/_form.html.haml [hunk 2]
```diff
@@ -0,0 +1,106 @@
+%div{"data-hook" => "admin_product_form_fields"}
```
Comment (luisramos0) [inline app/views/spree/admin/products/_form.html.haml]: data-hook should be removed right?

Comment (HugsDaniel) [reply app/views/spree/admin/products/_form.html.haml]: This data-hook is used for CSS : https://github.com/openfoodfoundation/spree/blob/41906362d931695e0616194341a68d2c4c85aaaf/backend/app/assets/stylesheets/admin/sections/_products.scss#L10

:/

**Comment by sauloperez** [comment]
> Maybe we should test 3050 and 3051 together in staging right?

I thought about the same but if an error is found, it'll make it harder to know its cause. I guess it doesn't take too long to test this page twice.

**Comment by sauloperez** [comment]
Staged on https://staging.katuma.org/

**Comment by RachL** [comment]
Everything looks good. However I don't understand your sentence @HugsDaniel 
Also, the field "Product Category" should now appear.  When did it disappeared ? :)
Testing notes :

https://docs.google.com/document/d/1vuuvGqHpnE95CdYQnEhv7qcixB_jn4vQga3EDVrqI08/edit#

**Pull request merged**